cdap-ui/app/cdap/components/OpsDashboard/RunsGraph/index.js

```diff
@@ -15,71 +15,50 @@
  */
 
 import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import {connect} from 'react-redux';
 import {renderGraph} from 'components/OpsDashboard/RunsGraph/graphRenderer';
-import IconSVG from 'components/IconSVG';
-import MyOperationsApi from 'api/operations';
-import {getCurrentNamespace} from 'services/NamespaceStore';
-import moment from 'moment';
-import {parseDashboardData} from 'components/OpsDashboard/RunsGraph/DataParser';
 import ToggleRunsList from 'components/OpsDashboard/RunsGraph/ToggleRunsList';
+import Legends from 'components/OpsDashboard/RunsGraph/Legends';
+import TypeSelector from 'components/OpsDashboard/RunsGraph/TypeSelector';
+import {Observable} from 'rxjs/Observable';
 
 require('./RunsGraph.scss');
 
 const RUNS_GRAPH_CONTAINER = 'runs-graph-container';
 
-export default class RunsGraph extends Component {
-  state = {
-    loading: true,
-    data: [],
-    pipelineCount: 0,
-    customAppCount: 0
+class RunsGraphView extends Component {
+  static propTypes = {
+    loading: PropTypes.bool,
+    data: PropTypes.array,
+    legends: PropTypes.object
   };
 
   componentDidMount() {
-    this.getData();
+    // update graph on window resize
+    this.windowResize$ = Observable.fromEvent(window, 'resize')
+      .debounceTime(500)
+      .subscribe(() => {
+        this.renderGraph();
+      });
   }
 
-  getData() {
-    let start = moment().subtract(24, 'h').format('x'),
-        duration = 1440;
-
-    start = parseInt(start, 10);
-
-    let params = {
-      start,
-      duration, // 24 hours in minutes
-      namespace: getCurrentNamespace()
-    };
-
-    MyOperationsApi.getDashboard(params)
-      .subscribe((res) => {
-        let {
-          pipelineCount,
-          customAppCount,
-          buckets
-        } = parseDashboardData(res, start, duration);
-
-        let data = Object.keys(buckets).map((time) => {
-          return {
-            ...buckets[time],
-            time
-          };
-        });
-
-        this.setState({
-          pipelineCount,
-          customAppCount,
-          data,
-          loading: false
-        });
-
-        // Render Graph
-        let containerElem = document.getElementById(RUNS_GRAPH_CONTAINER);
-
-        let width = containerElem.offsetWidth,
-            height = 230;
-        renderGraph('#runs-graph', width, height, this.state.data);
-      });
+  componentWillReceiveProps(nextProps) {
+    this.renderGraph(nextProps);
```

Comment:
Might want to optimize here if the props have actually changed. If not we could skip a re-render. Not the best optimization but anything we can do not re-render a complex graph might ease out the paint cycle.

Reply:
So the props being passed here is already limited, and any update to the props will require a change. There shouldn't be an update to the props where the data is not changed.